packages/@ember/-internals/glimmer/tests/integration/helpers/custom-helper-test.js

```diff
@@ -909,4 +923,95 @@ if (DEBUG) {
       }
     }
   );
+
+  moduleFor(
+    'Helpers test: argument-less helper invocation in named arguments position',
+    class extends RenderingTestCase {
+      constructor() {
+        super(...arguments);
+
+        this.registerComponent('bar', {
+          template: '[{{is-string @content}}][{{@content}}]',
+        });
+
+        this.registerHelper('is-string', ([value]) => typeof value === 'string');
+      }
+
+      ['@test invoking an argument-less helper without parens in named argument position is deprecated']() {
+        this.registerHelper('foo', () => 'Hello, world!');
+
+        expectDeprecation(
+          () => this.render('<Bar @content={{foo}} />', { foo: 'Not it!' }),
+          new RegExp(
+            /The `foo` helper was used in the `-top-level` template as /.source +
+              /`@content={{foo}}`\. This is ambigious between wanting the `@content` /.source +
+              /argument to be the `foo` helper itself, or the result of invoking the /.source +
+              /`foo` helper \(current behavior\)\. This implicit invocation behavior /.source +
+              /has been deprecated\./.source
+          )
+        );
+
+        this.assertText('[true][Hello, world!]');
+        this.assertStableRerender();
+      }
+
+      ['@test invoking an argument-less helper with parens in named argument position is not deprecated']() {
+        this.registerHelper('foo', () => 'Hello, world!');
+
+        expectNoDeprecation(() => this.render('<Bar @content={{(foo)}} />', { foo: 'Not it!' }));
+
+        this.assertText('[true][Hello, world!]');
+        this.assertStableRerender();
+      }
+
+      ['@test invoking an argument-less helper with quotes in named argument position is not deprecated']() {
+        this.registerHelper('foo', () => 'Hello, world!');
+
+        expectNoDeprecation(() => this.render('<Bar @content="{{foo}}" />', { foo: 'Not it!' }));
+
+        this.assertText('[true][Hello, world!]');
+        this.assertStableRerender();
+      }
+
+      ['@test passing a local helper in named argument position is not deprecated']() {
+        let foo = defineSimpleHelper(() => 'Hello, world!');
+
+        expectNoDeprecation(() =>
+          this.render(`{{#let this.foo as |foo|}}<Bar @content={{foo}} />{{/let}}`, { foo })
+        );
+
+        this.assertText('[false][Hello, world!]');
+        this.assertStableRerender();
+      }
+
+      // TODO: this one really should work, and there is a passing test in glimmer-vm,
+      // but somehow it doesn't work here. This is almost certainly a VM bug as something
+      // is trying to call `block.compile()` but `block` is the reference for `this.foo`.
+      // So the execution stack is probably off-by-one or something.
```

Comment:
cc @pzuraq, this seems mildly concerning, though unlikely to be introduced here and so don't seem like it would need to block merging. The test is ported from https://github.com/glimmerjs/glimmer-vm/blob/a19ce398de64077a52ebfae9f4878cec49a089c8/packages/%40glimmer/integration-tests/test/syntax/argument-less-helper-paren-less-invoke-test.ts#L66-L77, which does pass there.

Reply:
Actually – when I revert to `@glimmer/*` to 0.77.5, this particular test does pass, so there does appear to be a regression on 0.78.0. Do you think it is possible that https://github.com/glimmerjs/glimmer-vm/pull/1293 introduced  a stack bug here? I'm not sure why the same test on the vm side is not catching this, but perhaps it's a strict mode vs loose more thing?